pymeasure/instruments/keysight/keysight81160A.py

```diff
@@ -0,0 +1,208 @@
+    limit_state = Instrument.control(
+        ":VOLT{ch}:LIM:STAT?",
+        ":VOLT{ch}:LIM:STAT %s",
+        """ Control the limit state (string).""",
+        validator=strict_discrete_set,
+        map_values=True,
+        values={True: 1, "on": 1, "ON": 1, False: 0, "off": 0, "OFF": 0},
+        dynamic=True,
+    )
+
+    limit_high = Instrument.control(
+        ":VOLT{ch}:LIM:HIGH?",
+        ":VOLT{ch}:LIM:HIGH %f",
+        """ Control the high-level voltage limit (float).""",
+        validator=strict_range,
+        values=LIMIT_HIGH_RANGE,
+        dynamic=True,
+    )
+
+    limit_low = Instrument.control(
+        ":VOLT{ch}:LIM:LOW?",
+        ":VOLT{ch}:LIM:LOW %f",
+        """ Control the low-level voltage limit (float).""",
+        validator=strict_range,
+        values=LIMIT_LOW_RANGE,
+        dynamic=True,
+    )
+
+    memory_free = Instrument.measurement(
+        ":DATA{ch}:NVOL:FREE?",
+        """ Get the number of free non-volatile memory slots to store user waveforms (int).""",
+        dynamic=True,
+    )
+
+    waveforms = Instrument.measurement(
+        ":DATA{ch}:NVOL:CAT?",
+        """ Get the available user waveforms in memory (list[str]).""",
+        preprocess_reply=lambda v: v.replace('"', "").replace("\n", "").split(","),
+        dynamic=True,
+    )
+
+    volatile_waveform = Instrument.setting(
+        ":DATA{ch}:DAC VOLATILE, %s",
+        """ Set the volatile waveform data (str).""",
+        dynamic=True,
+    )
+
+    def save_waveform(self, waveform, name):
+        """
+        Save a waveform to the generator's nonvolatile memory.
+
+        :param waveform: The waveform data.
+        :param name: The name of the waveform.
+        """
+        self.volatile_waveform = waveform
+        self.write(f":DATA{self.id}:COPY {name}, VOLATILE")
+
+    def delete_waveform(self, name):
+        """
+        Delete a waveform from the generator's nonvolatile memory.
+
+        :param name: The name of the waveform.
+        """
+        self.write(f":DATA{self.id}:DEL {name.upper()}")
+
+
+class Keysight81160A(Agilent33500):
+    ch_1 = Instrument.ChannelCreator(Keysight81160AChannel, 1)
+    ch_2 = Instrument.ChannelCreator(Keysight81160AChannel, 2)
+
+    shape_values = WF_SHAPES
+    frequency_values = FREQUENCY_RANGE
+    amplitude_values = AMPLITUDE_RANGE
+    offset_values = OFFSET_RANGE
+    voltage_high_values = VOLTAGE_HIGH_RANGE
+    voltage_low_values = VOLTAGE_LOW_RANGE
+    square_dutycycle_values = SQUARE_DUTYCYCLE_RANGE
+    pulse_period_values = PULSE_PERIOD_RANGE
+    pulse_period_get_command = ":PULS:PER?"
+    pulse_period_set_command = ":PULS:PER %e"
+    pulse_width_values = PULSE_WIDTH_RANGE
+    pulse_dutycycle_values = PULSE_DUTYCYCLE_RANGE
+    pulse_transition_values = PULSE_TRANSITION_RANGE
+    pulse_transition_set_command = ":FUNC:PULS:TRAN %e"
+    burst_period_values = BURST_PERIOD
+    burst_ncycles_values = BURST_NCYCLES
+    arb_file_get_command = ":FUNC:USER?"
```

Comment:
As you just change the values (and a few commands), here it seems reasonable to inherit from the Agilent.